apps/history/declarativehistorymodel.cpp

```diff
@@ -66,6 +67,13 @@ void DeclarativeHistoryModel::remove(const QString &url)
     }
 }
 
+void DeclarativeHistoryModel::add(const QString &url, const QString &title)
+{
+    beginInsertRows(QModelIndex(), 0, 1);
```

Comment:
No need start model row insert here... search will query new set of history entries from database and sorting is handled by `updateModel`.

Just adding should be enough. There's no really means to checking history entries without  entering a search term.